dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -313,13 +315,58 @@ void convertOuterJoinToInnerJoin(List<TABLE_LIST>* join_list, TableOnExprList& t
   }
 }
 
-CalpontSystemCatalog::TableAliasName makeTableAliasName(TABLE_LIST* table)
+static execplan::Partitions getPartitions(TABLE* table)
+{
+  execplan::Partitions result;
+
+  if (table->part_info)
+  {
+    List_iterator<partition_element> part_el_it(table->part_info->partitions);
+
+    partition_element* pe;
+
+    while ((pe = part_el_it++)) // this is how server does it.
+    {
+      result.fPartNames.emplace_back(pe->partition_name);
+    }
+  }
+  return result;
+}
+static execplan::Partitions getPartitions(TABLE_LIST* table)
+{
+  execplan::Partitions result;
+
+  if (table->partition_names)
+  {
+    List_iterator<String> part_name_it(*(table->partition_names));
+
+    String* n;
+
+    while ((n = part_name_it++)) // this is how server does it.
+    {
+      std::string pn(n->ptr(), n->length());
+      result.fPartNames.push_back(pn);
+    }
+  }
+  return result;
+}
+
+
+CalpontSystemCatalog::TableAliasName makeTableAliasName_(TABLE_LIST* table)
```

Comment:
`makeTableAliasName_` is a peculiar choice TBH. Why do you need a prefix? 
If we adding `TableAliasName::partitions` attribute can we add another ctor that uses `getPartitions` internally?

Reply:
```If we adding TableAliasName::partitions attribute can we add another ctor that uses getPartitions internally?```

What do you mean by that?